getData.py

```diff
@@ -0,0 +1,131 @@
+            file.write(data)
+        else:
+            print("failed to get " + name)
+
+
+def create_score(symbol):
+    try:
+        filename = os.path.join('seperated', symbol, 'historical1.json')
+        with open(filename) as current:
+            filename = os.path.join('seperated', symbol, 'historical30.json')
+            with open(filename) as old:
+                data_current = json.load(current)
+                data_old = json.load(old)
+                score = data_current['high'] - data_old['high']
+                filename = os.path.join('seperated', symbol, 'score.json')
+                with open(filename, 'w') as score_file:
+                    data = {}
+                    data['score'] = score
+                    data['old'] = data_old['high']
+                    data['current'] = data_current['high']
+                    data['percentage'] = (((data_current['high'] - data['old']) /  data_current['high']) * 100)
+                    json.dump(data, score_file)
+    except:
+        pass
+
+
+def analyze_results(symbol):
+    if not os.path.exists('seperated'):
+        os.mkdir('seperated')
+    symbol_path = os.path.join('seperated', symbol)
+    if not os.path.exists(symbol_path):
+        os.mkdir(symbol_path)
+    create_separated_files(symbol, 'income')
+    create_separated_files(symbol, 'historical')
+    create_score(symbol)
+    # for filesname in os.listdir('raw'):
+    #     print(filesname)
+    # with open('dict.json', 'w') as w:
+    #     with open('testfile.json', 'r') as r:
+    #         data_list = json.load(r)
+    #         for d in data_list:
+    #             w.writelines('\n' + str(d['grossProfit']))
+
+
+def create_separated_files(symbol, querytype):
+    filename = os.path.join('raw', symbol + '_' + querytype + '.json')
+    if not os.path.exists(filename):
+        return
+    with open(filename) as f:
+        data_list = json.load(f)
+        i = 1
+        if querytype in data_list:
+            data_list = data_list[querytype]
+        for data in data_list:
+            filename_to_write = os.path.join('seperated', symbol, querytype + str(i) + '.json')
+            with open(filename_to_write, 'w') as file_to_write:
+                json.dump(data, file_to_write)
+            i += 1
+
+
+def get_conpanies(args):
+    with open(args.companies_file, 'r') as companies:
+        for symbol in companies:
+            if len(symbol) > 0:
+                symbol = symbol[:-1]
+                if args.fetchmode:
+                    print(symbol)
+                    urls = get_urls(symbol)
+                    get_results(symbol, urls)
+                else:
+                    analyze_results(symbol)
+
+
+def main():
+    args = parse_arguments()
+    get_conpanies(args)
+
+
+def parse_arguments():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--companies_file', type=str,
+                        help="the companies you want to download info for", required=True)
```

Comment:
* you can just name the argument without '--' if it is required
* `type=str` is the default
* `required=True` is default for non-flag syntax

```suggestion
    parser.add_argument('companies_file', help="file containing the companies you want to download info for")
```

(filetype)[https://docs.python.org/2/library/argparse.html#filetype-objects]